packages/edit-site/src/components/global-styles/hooks.js

```diff
@@ -0,0 +1,322 @@
+	// in which case we want the new array of values
+	// to override the old array (no merging).
+	if ( Array.isArray( srcValue ) ) {
+		return srcValue;
+	}
+}
+
+function mergeBaseAndUserConfigs( base, user ) {
+	return mergeWith( {}, base, user, mergeTreesCustomizer );
+}
+
+function addUserOriginToSettings( settingsToAdd ) {
+	PRESET_METADATA.forEach( ( { path } ) => {
+		const presetData = get( settingsToAdd, path );
+		if ( presetData ) {
+			set( settingsToAdd, path, {
+				user: presetData,
+			} );
+		}
+	} );
+	return settingsToAdd;
+}
+
+function removeUserOriginFromSettings( settingsToRemove ) {
+	PRESET_METADATA.forEach( ( { path } ) => {
+		const presetData = get( settingsToRemove, path );
+		if ( presetData ) {
+			set( settingsToRemove, path, ( presetData ?? {} ).user );
+		}
+	} );
+	return settingsToRemove;
+}
+
+function useGlobalStylesUserConfig() {
+	const { globalStylesId, content } = useSelect( ( select ) => {
+		const _globalStylesId = select( editSiteStore ).getSettings()
+			.__experimentalGlobalStylesUserEntityId;
+		return {
+			globalStylesId: _globalStylesId,
+			content: select( coreStore ).getEditedEntityRecord(
+				'postType',
+				'wp_global_styles',
+				_globalStylesId
+			)?.content,
+		};
+	}, [] );
+	const { getEditedEntityRecord } = useSelect( coreStore );
+	const { editEntityRecord } = useDispatch( coreStore );
+
+	const parseContent = ( contentToParse ) => {
+		let parsedConfig;
+		try {
+			parsedConfig = contentToParse ? JSON.parse( contentToParse ) : {};
+			// It is very important to verify if the flag isGlobalStylesUserThemeJSON is true.
+			// If it is not true the content was not escaped and is not safe.
+			if ( ! parsedConfig.isGlobalStylesUserThemeJSON ) {
+				parsedConfig = {};
+			} else {
+				parsedConfig = {
+					...parsedConfig,
+					settings: addUserOriginToSettings( parsedConfig.settings ),
+				};
+			}
+		} catch ( e ) {
+			/* eslint-disable no-console */
+			console.error( 'Global Styles User data is not valid' );
+			console.error( e );
+			/* eslint-enable no-console */
+			parsedConfig = {};
+		}
+
+		return parsedConfig;
+	};
+
+	const config = useMemo( () => {
+		return parseContent( content );
+	}, [ content ] );
+
+	const setConfig = useCallback(
+		( callback ) => {
+			const currentConfig = parseContent(
+				getEditedEntityRecord(
+					'postType',
+					'wp_global_styles',
+					globalStylesId
+				)?.content
+			);
+			const updatedConfig = callback( currentConfig );
+			editEntityRecord( 'postType', 'wp_global_styles', globalStylesId, {
+				content: JSON.stringify( {
+					...updatedConfig,
+					settings: removeUserOriginFromSettings(
+						updatedConfig.settings
+					),
+				} ),
+			} );
+		},
+		[ globalStylesId ]
+	);
+
+	return [ config, setConfig ];
+}
+
+function useGlobalStylesBaseConfig() {
+	const baseConfig = useSelect( ( select ) => {
+		return select( editSiteStore ).getSettings()
+			.__experimentalGlobalStylesBaseStyles;
```

Comment:
Working on other things I've noticed that we're dumping a lot of info we don't need at `__experimentalGlobalStylesBaseStyles`. It should only contain the base styles but not the settings and other properties of the `theme.json`.

I believe we may have done this because we also needed the base presets. Note that we have now available the presets for each origin (core, theme, user) under `__experimentalFeatures`, so `__experimentalGlobalStylesBaseStyles` can go back to styles-only.

Reply:
My idea at the moment is that we should remove both of them in favor of a `core-data` entity endpoint.